core/station/impl/src/factories/blockchains/ethereum.rs

```diff
@@ -0,0 +1,254 @@
+                .expect("failed to recover address");
+            let account_address = get_address_from_account(&account).await;
+            print(format!(
+                "test_recovered_address: {} {:?} {:?}",
+                hex::encode_prefixed(test_recovered_address).to_lowercase()
+                    == account_address.to_lowercase(),
+                hex::encode_prefixed(test_recovered_address),
+                account_address
+            ));
+        }
+
+        print(format!("signature: {:?}", signature));
+        let tx_signed = transaction.into_signed(signature);
+        let tx_envelope: alloy::consensus::TxEnvelope = tx_signed.into();
+        let tx_encoded = tx_envelope.encoded_2718();
+        print(format!("tx_encoded: {:?}", tx_encoded));
+
+        let sent_tx_hash = send_raw_transaction(&self.chain, &tx_encoded).await;
+        print(format!("sent_tx_hash: {:?}", sent_tx_hash));
+
+        Ok(BlockchainTransactionSubmitted {
+            details: vec![(
+                TRANSACTION_SUBMITTED_DETAILS_TRANSACTION_HASH_KEY.to_owned(),
+                sent_tx_hash,
+            )],
+        })
+    }
+}
+
+async fn ecdsa_pubkey_of(account: &Account) -> Vec<u8> {
+    let (key,) = ecdsa_public_key(EcdsaPublicKeyArgument {
+        canister_id: None,
+        derivation_path: principal_to_derivation_path(&account),
+        key_id: get_key_id(),
+    })
+    .await
+    .expect("failed to get public key");
+    key.public_key
+}
+
+async fn get_address_from_account(account: &Account) -> String {
+    let public_key = ecdsa_pubkey_of(&account).await;
+    let address = get_address_from_public_key(&public_key);
+    hex::encode_prefixed(&address)
+}
+
+fn get_address_from_public_key(public_key: &[u8]) -> Address {
+    let verifying_key = ecdsa::VerifyingKey::from_sec1_bytes(&public_key)
+        .expect("Failed to create VerifyingKey from public key bytes");
+    alloy::signers::utils::public_key_to_address(&verifying_key)
+}
+
+fn get_key_id() -> EcdsaKeyId {
+    // TODO: check what we should use as a name
+    let name: String = "dfx_test_key".to_string();
+
+    EcdsaKeyId {
+        curve: EcdsaCurve::Secp256k1,
+        name,
+    }
+}
+
+fn principal_to_derivation_path(account: &Account) -> Vec<Vec<u8>> {
+    let account_principal = Principal::from_slice(&account.id);
+    const SCHEMA: u8 = 1;
+    vec![vec![SCHEMA], account_principal.as_slice().to_vec()]
+}
+
+pub async fn send_raw_transaction(chain: &alloy_chains::Chain, raw_tx: &[u8]) -> String {
+    let config = None;
+    print(format!("getting services for chain: {:?}", chain));
+    let services = hashmap! {
+        alloy_chains::Chain::sepolia().id() => RpcServices::EthSepolia(Some(vec![EthSepoliaService::Alchemy])),
+        // alloy_chains::Chain::mainnet().id() => RpcServices::EthMainnet(None), // TODO: support mainnet
+    }.remove(&chain.id()).expect("chain not supported");
+
+    print(format!("got services"));
+
+    let cycles = 1000000000;
```

Comment:
estimate cycles here

Reply:
+1

Reply:
how to estimate cycles for a specific call?